Runtime/Scripts/RTPParameters.cs

```diff
@@ -1,22 +1,107 @@
+            if (parameter.hasValueScaleResolutionDownBy)
+                scaleResolutionDownBy = parameter.scaleResolutionDownBy;
+            rid = parameter.rid.AsAnsiStringWithFreeMem();
+        }
+
+        internal void CopyInternal(ref RTCRtpEncodingParametersInternal instance)
+        {
+            instance.active = active;
+            instance.hasValueMaxBitrate = maxBitrate.HasValue;
+            if(maxBitrate.HasValue)
+                instance.maxBitrate = maxBitrate.Value;
+            instance.hasValueMaxFramerate = maxFramerate.HasValue;
+            if (maxFramerate.HasValue)
+                instance.maxFramerate = maxFramerate.Value;
+            instance.hasValueScaleResolutionDownBy = scaleResolutionDownBy.HasValue;
+            if (scaleResolutionDownBy.HasValue)
+                instance.scaleResolutionDownBy = scaleResolutionDownBy.Value;
+            instance.rid = Marshal.StringToCoTaskMemAnsi(rid);
+        }
     }
 
-    internal struct RTCRtpEncodingParameters
+    public class RTCRtpSendParameters
     {
-        public bool active;
-        public ulong maxBitrate;
-        public string rid;
+        public string TransactionId => _transactionId;
+
+        public RTCRtpEncodingParameters[] Encodings => _encodings;
+
+
+        readonly RTCRtpEncodingParameters[] _encodings;
+        readonly string _transactionId;
+
+        internal RTCRtpSendParameters(RTCRtpSendParametersInternal parameters)
+        {
+            RTCRtpEncodingParametersInternal[] encodings = parameters.encodings.ToArray<RTCRtpEncodingParametersInternal>(parameters.encodingsLength);
+            _encodings = Array.ConvertAll(encodings, _ => new RTCRtpEncodingParameters(_));
+            _transactionId = parameters.transactionId.AsAnsiStringWithFreeMem();
+        }
+
+        internal IntPtr CreatePtr()
+        {
+            RTCRtpEncodingParametersInternal[] encodings = new RTCRtpEncodingParametersInternal[_encodings.Length];
+            for(int i = 0; i < _encodings.Length; i++)
+            {
+                _encodings[i].CopyInternal(ref encodings[i]);
+            }
+            RTCRtpSendParametersInternal instance = default;
+            instance.encodingsLength = _encodings.Length;
+            instance.encodings = IntPtrExtension.ToPtr(encodings);
+            instance.transactionId = Marshal.StringToCoTaskMemAnsi(_transactionId);
+            IntPtr ptr = Marshal.AllocCoTaskMem(Marshal.SizeOf(instance));
+            Marshal.StructureToPtr(instance, ptr, false);
+            return ptr;
+        }
+
+        static internal void DeletePtr(IntPtr ptr)
+        {
+            var instance = Marshal.PtrToStructure<RTCRtpSendParametersInternal>(ptr);
+            Marshal.FreeCoTaskMem(instance.encodings);
+            Marshal.FreeCoTaskMem(ptr);
+        }
     }
 
-    internal struct RTCRtpCodecParameters
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct RTCRtpSendParametersInternal
```

Comment:
I think it's better if we have this structure with the same name as the native one.

For the managed version, I think we can use another name,. such as `ManagedRTCRtpSendParameters`, or something.

Reply:
I agree with your suggestion that we have this structure in the managed code with the same name as the native one.

I am considering the name of the class in managed codes, this is the API for developers, the name is the same as the [javascript one](https://developer.mozilla.org/en-US/docs/Web/API/RTCRtpSendParameters). I think this is easier to understand.
https://github.com/Unity-Technologies/com.unity.webrtc/blob/a40fcde6349bb45a631f0ce4d6924e5802cc6878/Runtime/Scripts/RTPParameters.cs#L42

I guess we can change the name of the structure in the native code and the managed one, 
`RTCRtpSendParametersStruct` or `RTCRtpSendParametersData`.

https://github.com/Unity-Technologies/com.unity.webrtc/blob/a40fcde6349bb45a631f0ce4d6924e5802cc6878/Plugin~/WebRTCPlugin/WebRTCPlugin.cpp#L468-L473